### PR TITLE
Respond to polling station list call with 404 when the election does not exist

### LINKED ISSUE #167: Return 404 on when querying for non-existing elections
Right now an empty array gets returned, although a 404 response should be given and communicate an invalid `election_id` is given.

### PR DESCRIPTION
**Pull request by cikzh**
Closes #167

### CODE REVIEW

**Comment by cikzh** [comment]
I still need to change the OpenAPI spec, will put it out of Draft when this is done

**Comment by cikzh** [comment]
@jschuurk-kr FYI: this will also be rebased into #146

**Review by praseodym — APPROVED** [review]

File: backend/tests/polling_station_integration_test.rs
Comment (jschuurk-kr) [inline backend/tests/polling_station_integration_test.rs:1]: This files combines tests for two endpoints: `api/pollings_stations/<id>` and `api/polling_stations/<id>/data_entries/<id>`. I think we should split these across two files: `polling_station_integration_test.rs` and `data_entries_integration_test.rs`.

**Review by rnijveld — APPROVED** [review]

File: backend/tests/polling_station_integration_test.rs
Comment (cikzh) [reply backend/tests/polling_station_integration_test.rs:1]: The code that is tested by this file lives in the same Rust module, so I think it makes sense to keep this in the same file.

Comment (jschuurk-kr) [reply backend/tests/polling_station_integration_test.rs:1]: I'd split integration tests based on the endpoints, not on the files. We're testing from the outside, so it doesn't matter for the tests which files with code are involved.

Relatedly, it looks like the tests are currently sorted on success vs error first, and then on endpoint. That does not map with how I think about integration tests and their coverage. Not that I can't learn a new habit, of course. :-) So glad to have this discussion here, so we an make a decision and then apply it consistently.

Comment (cikzh) [reply backend/tests/polling_station_integration_test.rs:1]: Shall we create an issue for that? Then we can keep this PR about the 404 response

Comment (jschuurk-kr) [reply backend/tests/polling_station_integration_test.rs:1]: I was hoping we'd be able to come to a decision fast enough for that not to be necessary. Let's create a different issue for this if we don't have a decision after standup on Tuesday?

Comment (praseodym) [reply backend/tests/polling_station_integration_test.rs:1]: As discussed in the standup: we split the integration tests into a file per endpoint, not necessarily following the module structure. @rnijveld will create a PR after this one is merged to reorganise the current tests.

**Pull request merged**